Local storage

HanaSabih opened the pull request:
### preserve data in the browser

- I have created an array dataForm 
- if the form is working correctly and then submit  
- the data saved to the local storage.
- When the user loads the page if there is any data in the local storage the input fields are pre-filled with this data.
- I created a single JavaScript object  called (newDataForm) with all the data from the entire form and saved it in local storage

HanaSabih committed:
local storage

Thoth1111 committed:
Modify index.js

HanaSabih committed:
Update index.js

GedeonTS reviewed (changes requested):
# INVALID REVIEW REQUEST

Hi @HanaSabih , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
<Img src="https://media.giphy.com/media/11tVh1XRNAunYI/giphy.gif">
## Invalid Code Review Request

You have submitted a project, that has a PR that is not descriptive enough, kindly  add more details to it, I suggest using bullets for each action you did following the project requirements

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

franclobo commented on index.js:
```diff
@@ -148,5 +159,13 @@ gontactForm.addEventListener('submit', (e) => {
   } else {
     messageform.style.display = 'block';
     gontactForm.submit();
+    // =========local storage=========
+    const newDataForm = {
+      name: name.value,
+      email: email.value,
+      msg: msg.value,
+    };
+    dataForm.push(newDataForm);
+    localStorage.setItem('info', JSON.stringify(dataForm));
```
- [ ] Please 🙏🏽 for a better user experience the form should do the following:

> The contact form information is persistent to page reloads.
> After filling the form, **and not submitting it**, the entered values should be saved in the browser.
> When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page.

franclobo reviewed (changes requested):
Hi @HanaSabih, 

# Required Changes ♻️
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Correct Github flow ✔️ 
- Descriptive PR ✔️ 
- Professional README file ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HanaSabih committed:
fix all required changes

The contact form information is persistent to page reloads.
After filling out the form, and not submitting it, the entered values should be saved in the browser.
When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page.

Gambit142 commented on index.js:
```diff
@@ -134,9 +134,29 @@ document.querySelectorAll('.btnModal').forEach((n) => n.addEventListener('click'
+const name = document.getElementById('name');
+const msg = document.getElementById('msg');
+
+const dataForm = JSON.parse(localStorage.info) || [];
+
```
- [ ] Great work attempting to make the changes suggested by the previous reviewer. However, the feature has not been implemented and still doesn't work. Remember that the requirements state that you should save all form data in an object not an array and save the object in the localStorage. Kindly implement this. 👍 

#### Hint: Delete this line of code

Gambit142 commented on index.js:
```diff
@@ -134,9 +134,29 @@ document.querySelectorAll('.btnModal').forEach((n) => n.addEventListener('click'
+    email: email.value,
+    msg: msg.value,
+  };
+  dataForm.push(newDataForm);
+  localStorage.setItem('info', JSON.stringify(dataForm));
```
- [ ] Kindly store an `object` and not an `array` in the `localStorage`. You don't need the `dataForm` array. 👍 

#### Hint: `localStorage.setItem('info', JSON.stringify(newDataForm))`

Gambit142 commented on index.js:
```diff
@@ -150,3 +170,10 @@ gontactForm.addEventListener('submit', (e) => {
   }
 });
+
+// display changes
+dataForm.forEach((inputInfo) => {
+  name.value = inputInfo.name;
+  email.value = inputInfo.email;
+  msg.value = inputInfo.msg;
+});
```
- [ ] To make it work, you need to get the data from the `localStorage` and display the data in the several form input. 👍 

#### Hint: `const obtainedData = JSON.parse(localStorage.getItem('info'));`. Then you assign the elements with the values from the `obtainedData.` 👍

Gambit142 reviewed (changes requested):
Hi @HanaSabih @Thoth1111 👋, 

Great work on making the changes requested by a previous reviewer 👏 

## Highlights
- All linters checks passed ✔️ 
- Pull request summary is descriptive ✔️ 

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HanaSabih committed:
fix the object

binhussen reviewed (approved):
Hi @HanaSabih , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
<img src="https://user-images.githubusercontent.com/65852621/133696182-6c5b784b-0083-427a-b24d-3ec870915f79.jpg"/>
## Highlights
- All linter checks passed ✔️
- Descriptive pull request ✔️
- Used GitHub flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
